### PR TITLE
fix and update experimental Next.js template to work on `@opennextjs/cloudflare@0.3.x`

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
Changes made in `@opennextjs/cloudflare@0.3.0` haven't been updated in the C3 experimental template causing C3 to currently create broken open-next applications, this PR is addressing that plus some other small minor related  updates.

---

- Tests
  - [ ] TODO (before merge)
  - [ ] Tests included
  - [x] Tests not necessary because: the tests are already present
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required
  - [ ] Not required because:
- Public documentation
  - [ ] TODO (before merge)
  - [x] Cloudflare docs PR(s): https://github.com/cloudflare/cloudflare-docs/pull/18946
  - [ ] Documentation not necessary because:

### CODE REVIEW

File: packages/create-cloudflare/templates-experimental/next/templates/.gitignore [hunk 1]
```diff
@@ -35,8 +35,5 @@ yarn-error.log*
 *.tsbuildinfo
 next-env.d.ts
 
-
-# Cloudflare related
-/.save.next
-/.worker-next
-/.wrangler
```
Comment (dario-piotrowicz) [inline packages/create-cloudflare/templates-experimental/next/templates/.gitignore:42]: We always add `.wrangler` to `.gitignore` files anyways: https://github.com/cloudflare/workers-sdk/blob/998902238552c860cf9dd9f629e8f351352ae3c4/packages/create-cloudflare/src/templates.ts#L802
so there's no reason to do that here as well

File: packages/create-cloudflare/templates-experimental/next/templates/.gitignore [hunk 2]
```diff
@@ -35,8 +35,5 @@ yarn-error.log*
 *.tsbuildinfo
 next-env.d.ts
 
-
-# Cloudflare related
-/.save.next
```
Comment (dario-piotrowicz) [inline packages/create-cloudflare/templates-experimental/next/templates/.gitignore:40]: we don't have a `.save.next` dir anymore

File: packages/create-cloudflare/templates-experimental/next/templates/.gitignore [hunk 3]
```diff
@@ -35,8 +35,5 @@ yarn-error.log*
...
-/.worker-next
-/.wrangler
+# OpenNext
```
Comment (dario-piotrowicz) [inline packages/create-cloudflare/templates-experimental/next/templates/.gitignore:38]: @vicb is there any new open-next file/directory that we should include here?

File: packages/create-cloudflare/templates-experimental/next/templates/wrangler.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #:schema node_modules/wrangler/config-schema.json
 name = "<app-name>"
-main = ".worker-next/index.mjs"
+main = ".open-next/index.mjs"
```
Comment (vicb) [inline packages/create-cloudflare/templates-experimental/next/templates/wrangler.toml:3]: It should be worker.js

**Review by vicb — CHANGES_REQUESTED** [review]
Thanks

Could you please also add a .dev.vars, using https://github.com/flarelabs-net/workers-next/blob/main/.dev.vars.example as a template

File: packages/create-cloudflare/templates-experimental/next/templates/.gitignore (hunk 3 above)
Comment (dario-piotrowicz) [reply packages/create-cloudflare/templates-experimental/next/templates/.gitignore:38]: yes I think it's ok too, I just wanted to double check, thanks 🙂

**Review by vicb — APPROVED** [review]
Thanks.

Maybe create an issue on https://github.com/opennextjs/docs to update the docs when this PR is released to c3

**Comment by dario-piotrowicz** [comment]
> Thanks.
> 
> Maybe create an issue on https://github.com/opennextjs/docs to update the docs when this PR is released to c3

This is the Next.js guide that uses the open-next adapter: https://developers.cloudflare.com/workers/frameworks/framework-guides/nextjs/

I don't think there's anything in the changes I'm making that needs updating there (otherwise I'd do it right away)

**Review by edmundhung — APPROVED** [review]

**Comment by vicb** [comment]
> > Thanks.
> > Maybe create an issue on https://github.com/opennextjs/docs to update the docs when this PR is released to c3
> 
> This is the Next.js guide that uses the open-next adapter: https://developers.cloudflare.com/workers/frameworks/framework-guides/nextjs/
> 
> I don't think there's anything in the changes I'm making that needs updating there (otherwise I'd do open a PR right away and merge it after the C3 release)

We need to update [that](https://github.com/opennextjs/docs/blob/main/pages/cloudflare/get-started.mdx#:~:text=npm%20create%20cloudflare%40latest%20%2D%2D%20%2D%2Dtemplate%20https%3A//github.com/flarelabs%2Dnet/workers%2Dnext) which is repeated multiple times to use the c3 experimental template

**Comment by dario-piotrowicz** [comment]
> > > Thanks.
> > > Maybe create an issue on https://github.com/opennextjs/docs to update the docs when this PR is released to c3
> > 
> > 
> > This is the Next.js guide that uses the open-next adapter: https://developers.cloudflare.com/workers/frameworks/framework-guides/nextjs/
> > I don't think there's anything in the changes I'm making that needs updating there (otherwise I'd do open a PR right away and merge it after the C3 release)
> 
> We need to update [that](https://github.com/opennextjs/docs/blob/main/pages/cloudflare/get-started.mdx#:~:text=npm%20create%20cloudflare%40latest%20%2D%2D%20%2D%2Dtemplate%20https%3A//github.com/flarelabs%2Dnet/workers%2Dnext) which is repeated multiple times to use the c3 experimental template

ah ok, sorry I thought we were talking about the Cloudflare docs 👍

**Comment by dario-piotrowicz** [comment]
@vicb https://github.com/opennextjs/docs/pull/44 👍

**Comment by dario-piotrowicz** [comment]
Merging this PR as the only failing tests are e2es (both experimental and not) running nuxt using yarn under ubuntu and:
 - this PR is completely unrelated to nuxt
 - I can see the same failures in other existing PRs

**Pull request merged**